compiler/rustc_infer/src/infer/error_reporting/nice_region_error/static_impl_trait.rs

```diff
@@ -353,12 +356,49 @@ pub fn suggest_new_region_bound(
                     .is_some()
                 {
                 } else {
-                    err.span_suggestion_verbose(
-                        fn_return.span.shrink_to_hi(),
-                        &format!("{declare} `{ty}` {captures}, {explicit}",),
-                        &plus_lt,
-                        Applicability::MaybeIncorrect,
-                    );
+                    if tcx
+                        .hir()
+                        .get_generics(scope_def_id)
+                        .map(|generics| {
+                            let mut lt_spans: Vec<Span> = vec![];
+                            for p in generics.params {
+                                if p.is_elided_lifetime() {
+                                    lt_spans.push(p.span.shrink_to_hi());
+                                }
+                            }
+                            if lt_spans.len() > 1 {
+                                // more than one elided lifetimes found in inputs, can't use explicit `'_` lifetime,
+                                // use a named lifetime parameter `'a` instead.
+                                let mut spans_suggs: Vec<_> = vec![];
+                                for lt_span in lt_spans {
+                                    spans_suggs.push((lt_span.shrink_to_hi(), "'a ".to_string()));
+                                }
+                                spans_suggs
+                                    .push((generics.span.shrink_to_hi(), "<'a>".to_string()));
+                                spans_suggs
+                                    .push((fn_return.span.shrink_to_hi(), " + 'a ".to_string()));
+                                err.multipart_suggestion_verbose(
+                                    &format!(
+                                        "{declare} `{ty}` {captures}, {}",
+                                        "you can consider introducing a named lifetime parameter `'a`",
+                                    ),
+                                    spans_suggs,
+                                    Applicability::MaybeIncorrect,
+                                );
+                                false
+                            } else {
+                                true
+                            }
+                        })
+                        .unwrap()
```

Comment:
Don't use `unwrap()`. You can use let chains like this (modulo formatting):

```rust
if let Some(generics) = tcx.hir().get_generics(scope_def_id)
    && let lt_spans = generics
        .params
        .iter()
        .filter(|p| p.is_elided_lifetime())
        .map(|p| p.span.shrink_to_hi())
        .collect::<Vec<_>>()
    && lt_spans.len() > 1
{
    let span_suggs: Vec<_> = lt_spans.into_iter().map(|sp| (sp, "'a ".to_string())).collect();
    ///...
} else {
    err.span_suggestion_verbose( // already existing sugg
}
```

Reply:
Thanks for great suggestions, which made the updated code much more elegant!